### PR TITLE
WIP:Added fss storage module independent of OKE network module

### PR DESCRIPTION
**Pull request by karthicgit**
PR includes fss module changes ,docs updated for fss , encryption variable for cluster renamed.

### CODE REVIEW

File: docs/instructions.adoc [hunk 1]
```diff
@@ -74,8 +74,8 @@ If you wish to use {uri-oci-kms}[OCI KMS] to encrypt Kubernetes secrets, the fol
 * link:#adding-the-bastion-host[bastion must be enabled]
 * link:#enabling-instance_principal-on-the-operator-host[operator instance_principal must be enabled]
 
-* use_encryption must be set to _true_
-* kms_key_id must be provided
+* use_cluster_encryption must be set to _true_
```
Comment (hyder) [inline docs/instructions.adoc:77]: Please use backticks when documenting variables: `my_variable`

File: docs/dependencies.adoc [hunk 2]
```diff
@@ -46,7 +46,7 @@ The following table documents the {uri-terraform-options}[Terraform Options] dep
 |Whether to create an authentication secret for OCIR
 |create_bastion_host = true, create_operator = true, admin_instance_principal = true, secret_id = secret ocid
 
-|use_encryption
+|use_cluster_encryption
 |Uses OCI KMS to encrypt data in OKE's underlying etcd
 |create_bastion_host = true, create_operator = true, admin_instance_principal = true
```
Comment (hyder) [inline docs/dependencies.adoc:51]: Can you please rebase?

File: docs/terraformoptions.adoc [hunk 3]
```diff
@@ -707,6 +707,42 @@ node_pools = {
...
+== File Storage
+
+[stripes=odd,cols="1m,4d,2m,2m", options=header,width="100%"] 
+|===
+|Parameter
+|Description
+|Values
+|Default
+
+|enable_fss
```
Comment (hyder) [inline docs/terraformoptions.adoc:719]: Variable name should be create_fss

File: docs/instructions.adoc (hunk 1 above)
Comment (karthicgit) [reply docs/instructions.adoc:77]: ok.

File: main.tf [hunk 4]
```diff
@@ -255,6 +255,37 @@ module "oke" {
...
+
+  # Export set configuration
+  max_fs_stat_bytes = var.max_fs_stat_bytes
+  max_fs_stat_files = var.max_fs_stat_files
+
+  providers = {
+    oci.home = oci.home
+  }
+
+  count = var.enable_fss == true ? 1 : 0
```
Comment (hyder) [inline main.tf:285]: create_fss as above

File: modules/storage/locals.tf [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+
+  #IANA protocol numbers
+  tcp_protocol = 6
+
+  udp_protocol = 17
+
+  # ad_names = [
```
Comment (hyder) [inline modules/storage/locals.tf:11]: if it's not needed, please remove

File: terraform.tfvars.example [hunk 6]
```diff
@@ -190,6 +190,14 @@ preferred_load_balancer      = "public"
 # public_lb_allowed_cidrs    = ["0.0.0.0/0"] # By default, anywhere i.e. 0.0.0.0/0 is allowed
 # public_lb_allowed_ports    = [443,"9001-9002"] # By default, only 443 is allowed
 
+#fss
+enable_fss = false
```
Comment (hyder) [inline terraform.tfvars.example:195]: create_fss

**Review by hyder — CHANGES_REQUESTED** [review]
For the NSG, you should consider how to add the FSS rules for the existing worker NSG. You can either add them as conditional rules if `create_fss` is true or use the NSG chaining method that @agregory999 was suggesting in #425

File: docs/instructions.adoc (hunk 1 above)
Comment (karthicgit) [reply docs/instructions.adoc:77]: backtick wont work in this place. Backtick was not there as well before its just a variable name change.

File: modules/storage/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+output "fss_mount_target_id" {
+  value = oci_file_storage_mount_target.fss_mount_target.id
+}
+
+output "fss_subnet_id" {
+  value = (var.create_fss == true) ? oci_core_subnet.fss.id : ""
+}
+
+output "fss_mount_target_nsg_id" {
+  value = (var.create_fss == true) ? oci_core_network_security_group.fss_mt.id : ""
```
Comment (hyder) [inline modules/storage/outputs.tf:14]: Put the conditional on the module definition itself. Then, we don't need to check for every resource

Comment (karthicgit) [reply modules/storage/outputs.tf]: done.

File: main.tf [hunk 8]
```diff
@@ -255,6 +259,36 @@ module "oke" {
...
+  compartment_id      = var.compartment_id
+  availability_domain = var.availability_domains["fss"]
+  label_prefix        = var.label_prefix
+
+  # FSS network information
+  subnets      = var.subnets
+  vcn_id       = module.vcn.vcn_id
+  nat_route_id = module.vcn.nat_route_id
+
+  create_fss     = var.create_fss
```
Comment (hyder) [inline main.tf:277]: passing this parameter is not needed since you're making the entire module conditional.

File: terraform.tfvars.example [hunk 9]
```diff
@@ -54,6 +54,7 @@ subnets = {
   int_lb   = { netnum = 16, newbits = 11 }
   pub_lb   = { netnum = 17, newbits = 11 }
   workers  = { netnum = 1, newbits = 2 }
+  fss  = { netnum = 18, newbits = 11 }
```
Comment (hyder) [inline terraform.tfvars.example:57]: formatting

File: terraform.tfvars.example [hunk 10]
```diff
@@ -119,7 +120,8 @@ operator_notification_topic    = ""
 # availability_domains
 availability_domains = {
   bastion  = 1,
-  operator = 1
+  operator = 1,
+  fss = 1
```
Comment (hyder) [inline terraform.tfvars.example:124]: formatting

File: main.tf (hunk 8 above)
Comment (karthicgit) [reply main.tf]: done

**Comment by hyder** [comment]
To be merged with #491

**Pull request closed without merging**